### DISCUSSION TITLE
Install t8code on CentOS 7.9

### DISCUSSION THREAD

**Discussion by jmark** [DLR-AMR/t8code#784] [Setup & Installation]
This time, I tried to install t8code on our cluster, based on CentOS 7.9 with kernel version 3.10.0-1160.el7.x86_64. I loaded the required modules:
```bash
$ module list
Currently Loaded Modulefiles:
  1) tools/automake/1.16.1
  2) tools/git/2.37.1
  3) tools/cmake/3.23.0
  4) compiler/gcc/9.4.0
  5) mpi/openmpi/4.1.1_gcc94
  6) tools/libtool/2.4.6
```
but the building failed (not a VTK error).

To find the cause of the error, could you tell me what toolchain you use on your clusters? I saw in the [Wiki](https://github.com/DLR-AMR/t8code/wiki/Setting-up-t8code-on-JUWELS-(or-other-Slurm-based-systems)#required-modules) that it is an Intel compiler and MPI. Strangely, on my Fedora desktop, it worked fine with the default GCC 7.3.1.

_Originally posted by @CsatiZoltan in https://github.com/DLR-AMR/t8code/discussions/782#discussioncomment-7194917_

**Comment by jmark** [comment]
@CsatiZoltan The wiki lists the toolchain. Of course, it differs slightly from cluster to cluster.
Could you post the error message you get?

**Reply by CsatiZoltan** [reply]
> The wiki lists the toolchain.

Do you mean the [requirements](https://github.com/DLR-AMR/t8code/wiki/Installation#requirements)? If so, they are met in our case.

I attached in a zip file the outputs. There are some errors during the configuration, but what I was rather interested in was the difference between the successful compilation on Fedora (#782) and on this CentOS based cluster.
The extra error I found:
```
configure:22106: error: ../../t8code/sc/configure failed for sc
```
which does not tell much. I continue experimenting with the compilation.
[logs.zip](https://github.com/DLR-AMR/t8code/files/12817840/logs.zip)

**Reply by jmark** [reply]
It seems that the configuration for `libsc`, a dependency of `t8code`, is failing. Maybe you could try to get `libsc` working in isolation on your cluster to find out where the problem lies.

This is the version the latest t8code sources point to:
https://github.com/cburstedde/libsc/tree/7d30c6b6af813b6f3cd08b891ba417a832cd505f

**Comment by cburstedde** [comment]
We're generally trying to keep the develop branch of the submodules p4est and sc current and suitable for t8code. When in doubt, running git submodule update on t8code should check out versions that work.

Wrt. t8code, p4est and sc are subdirectories parallel to each other. When checking out p4est standalone, sc in turn lives in a subdirectory that can be checked out with git submodule init and update from p4est. sc, otoh, builds standalone as well. We'll be happy to help out building the submodules on this platform.
